src/main/java/org/testng/TestRunner.java

```diff
@@ -678,7 +680,6 @@ public int compareTo(@Nonnull IWorker<ITestNGMethod> other) {
         });
 
     runJUnitWorkers(workers);
-    m_allTestMethods = runMethods.toArray(new ITestNGMethod[0]);
```

Comment:
`runMethods` is no more used: it was useless before or a feature is now broken.

Reply:
Proposal: keep usage of `m_allTestMethods` as a cache when the JUnit feature is used.

Reply:
Fixed